processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -302,6 +302,7 @@ else if ( mapping.getSourceName() != null ) {
                                     .qualifiers( mapping.getQualifiers() )
                                     .resultType( mapping.getResultType() )
                                     .dateFormat( mapping.getDateFormat() )
+                                    .numberFormat( mapping != null ? mapping.getNumberFormat() : null)
```

Comment:
The null check isn't really needed here. If it was null, an NPE would have been raised before.